weave/conftest.py

```diff
@@ -176,15 +176,38 @@ def __getattribute__(self, name):
         if callable(attr) and not name.startswith("_") and name != "flush":
 
             def wrapper(*args, **kwargs):
-                res = attr(*args, **kwargs)
                 self_super._flush()
+                res = attr(*args, **kwargs)
```

Comment:
This was just a minor change to the test class - not specifically related to this PR, but i noticed the issue while debugging. Should flush first